openmdao/components/interp_util/interp.py

```diff
@@ -71,83 +82,175 @@ class InterpND(object):
-                raise ValueError("The points in dimension %d must be strictly "
-                                 "ascending" % i)
-            if not np.asarray(p).ndim == 1:
-                raise ValueError("The points in dimension %d must be "
-                                 "1-dimensional" % i)
-            if not values.shape[i] == n_p:
-                raise ValueError("There are %d points and %d values in "
-                                 "dimension %d" % (len(p), values.shape[i], i))
+            if isinstance(points, np.ndarray):
+                points = [points]
+
+            for i, p in enumerate(points):
+                n_p = len(p)
+                if not np.all(np.diff(p) > 0.):
+                    raise ValueError("The points in dimension %d must be strictly "
+                                     "ascending" % i)
+                if not np.asarray(p).ndim == 1:
+                    raise ValueError("The points in dimension %d must be "
+                                     "1-dimensional" % i)
+                if values is not None and not values.shape[i] == n_p:
+                    raise ValueError("There are %d points and %d values in "
+                                     "dimension %d" % (len(p), values.shape[i], i))
 
         self.grid = tuple([np.asarray(p) for p in points])
         self.values = values
+        self.x_interp = x_interp
+
         self._xi = None
         self._d_dx = None
         self._d_dgrid = None
         self._d_dvalues = None
-        self.training_data_gradients = False
+        self._compute_d_dvalues = False
 
         # Cache spline coefficients.
-        interp = INTERP_METHODS[interp_method]
+        interp = INTERP_METHODS[method]
 
-        if interp_method.startswith('scipy'):
-            kwargs['interp_method'] = interp_method
+        if method.startswith('scipy'):
+            kwargs['interp_method'] = method
 
-        table = interp(self.grid, self.values, interp, **kwargs)
+        table = interp(self.grid, values, interp, **kwargs)
         table.check_config()
         self.table = table
         self._interp = interp
         self._interp_options = kwargs
 
-    def interpolate(self, xi):
+    def interpolate(self, x, compute_derivative=False):
         """
         Interpolate at the sample coordinates.
 
+        Parameters
+        ----------
+        x : ndarray of shape (..., ndim)
+            Location to provide interpolation.
+        compute_derivative : bool
+            Set to True to compute derivatives with respect to x.
+
+        Returns
+        -------
+        ndarray
+            Value of interpolant at all sample points.
+        ndarray
+            Value of derivative of interpolated output with respect to input x. (Only when
+            compute_derivative is True.)
+        """
+        xnew = self._interpolate(np.atleast_1d(x))
+
+        if compute_derivative:
+            return xnew, self._d_dx
+        else:
+            return xnew
```

Comment:
There isn't a danger, since they are computed during interpolation. However, I also need to set the proper flags for derivatives computation so that it doesn't waste time computing them when you don't want them, and it looks like I didn't do it here.  I have fixed this.